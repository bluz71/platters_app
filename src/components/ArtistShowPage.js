import React, { Component } from 'react';

class ArtistShowPage extends Component {
  constructor(props) {
    super(props);

    this.artistSlug = props.match.params.id;
  }

  render() {
    return (
      <h1>Artist Show Page for {this.artistSlug}</h1>
    );
  }
}

export default ArtistShowPage;
