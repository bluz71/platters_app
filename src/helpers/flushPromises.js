// Details:
//   https://blog.rescale.com/testing-promise-side-effects-with-asyncawait
//   https://github.com/facebook/jest/issues/2157#issuecomment-279171856

const flushPromises = () => new Promise(resolve => setImmediate(resolve));

export default flushPromises;
