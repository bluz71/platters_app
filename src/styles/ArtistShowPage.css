.ArtistShow a {
  text-decoration: none;
}

.ArtistShow .description p {
  display: inline;
  color: #515151;
}

.ArtistShow .website {
  margin-top: 1em;
  margin-bottom: 1em;
}

/* #artist { */

/*   .albums-order { */
/*     text-align: center; */
/*     font-size: 0.9em; */

/*     li { */
/*       display: inline; */
/*       margin: 0 1em; */
/*       font-weight: lighter; */

/*       a { */
/*         color: $grey-dark; */

/*         &.active { */
/*           color: $green-yeti; */
/*           font-weight: bold; */
/*         } */

/*         &:hover { */
/*           color: $green-yeti; */
/*         } */
/*       } */
/*     } */
/*   } */

/*   .album { */
/*     border-bottom: 1px solid $grey-mid-light; */

/*     img { */
/*       height: 80px; */
/*       width: 80px; */
/*       float: left; */
/*       border: 1px solid $grey-mid-dark; */
/*       border-radius: 50%; */
/*       margin-right: 1em; */

/*       &:hover { */
/*         border: 1px solid $green-yeti; */
/*       } */
/*     } */

/*     .title { */
/*       height: 2.1em; */
/*       white-space: nowrap; */
/*       overflow: hidden; */
/*       text-overflow: ellipsis; */
/*       margin: 1em 0 0; */

/*       h2 { */
/*         display: inline; */
/*         font-size: 1.8em; */
/*       } */

/*       @media (max-width: $screen-sm-max) { */
/*         .icon { */
/*           display: none; */
/*         } */
/*       } */

/*       @media (min-width: $screen-md-min) { */
/*         .icon { */
/*           margin-top: 1em; */
/*           float: right; */
/*           font-size: 0.8em; */
/*           color: $grey-dark; */
/*         } */
/*       } */
/*     } */

/*     .album-details { */
/*       margin-top: 0; */

/*       small { */
/*         font-size: 0.45em; */
/*       } */
/*     } */

/*     a { */
/*       color: $grey-dark; */
/*       text-decoration: none; */

/*       &:hover { */
/*         color: $green-yeti; */
/*       } */
/*     } */

/*     ul { */
/*       font-size: 0.8em; */
/*       color: $grey-dark; */
/*       padding-left: 0; */
/*       white-space: nowrap; */
/*       overflow: hidden; */
/*       text-overflow: ellipsis; */

/*       li { */
/*         display: inline; */
/*         list-style-type: none; */
/*       } */
/*     } */
/*   } */
/* } */
