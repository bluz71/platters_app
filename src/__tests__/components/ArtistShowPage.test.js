import React from 'react';
import { shallow } from 'enzyme';
import flushPromises from '../../helpers/flushPromises';
import ArtistShowPage from '../../components/ArtistShowPage';

describe('<ArtistShowPage />', () => {
  it('renders artist show page', async () => {
    const match = { params: { id: 'abc' } };
    const wrapper = shallow(<ArtistShowPage match={match} location />);
    await flushPromises();
    wrapper.update();
    expect(wrapper).toMatchSnapshot();
  });
});
