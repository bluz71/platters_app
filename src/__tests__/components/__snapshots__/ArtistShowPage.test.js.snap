// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ArtistShowPage /> renders artist show page 1`] = `
<Row
  bsClass="row"
  componentClass="div"
>
  <Col
    bsClass="col"
    className="ArtistShow"
    componentClass="div"
    md={10}
    mdOffset={1}
  >
    <PageHeader
      bsClass="page-header"
    >
      ABC
    </PageHeader>
    <div
      className="description"
    >
      <p>
        ABC Band
         
      </p>
      <a
        href="https://www.wikipedia.org/wiki/abc_band"
        rel="noopener noreferrer"
        target="_blank"
      >
        Wikipedia
      </a>
      <div
        className="website"
      >
        <span
          className="icon spacer-right-xxs"
        >
          <FontAwesome
            name="globe"
          />
        </span>
        <a
          href="https://abc_band.com"
          rel="noopener noreferrer"
          target="_blank"
        >
          abc_band.com
        </a>
      </div>
    </div>
    <div
      className="albums"
    >
      <PageHeader
        bsClass="page-header"
      >
        Albums 
        <small>
          (
          2
           
          Albums
          )
        </small>
      </PageHeader>
      <div>
        <ul
          className="albums-order"
        >
          <li
            className="active"
            onClick={[Function]}
          >
            Newest
          </li>
          <li
            onClick={[Function]}
          >
            Oldest
          </li>
          <li
            onClick={[Function]}
          >
            Longest
          </li>
          <li
            onClick={[Function]}
          >
            Name
          </li>
        </ul>
        <ArtistAlbumsList
          albums={
            Array [
              Object {
                "comments_count": 20,
                "cover_url": "http://localhost:3000/uploads/album/cover/1/small_ABC--DEF.jpg",
                "genre": "Rock",
                "id": 1,
                "slug": "def",
                "title": "DEF",
                "total_duration": "45:23",
                "tracks_count": 10,
                "tracks_summary": Array [
                  "1. First",
                  "2. Second",
                  "3. Third",
                  "4. Fourth",
                  "5. Fifth",
                  "6. Sixth",
                  "...",
                ],
                "year": 2000,
              },
              Object {
                "comments_count": 9,
                "cover_url": "http://localhost:3000/uploads/album/cover/2/small_ABC--XYZ.jpg",
                "genre": "Rock",
                "id": 2,
                "slug": "xyz",
                "title": "XYZ",
                "total_duration": "50:23",
                "tracks_count": 10,
                "tracks_summary": Array [
                  "1. First",
                  "2. Second",
                  "3. Third",
                  "4. Fourth",
                  "5. Fifth",
                  "6. Sixth",
                  "...",
                ],
                "year": 1995,
              },
            ]
          }
          artistSlug="abc"
          onGenre={[Function]}
          onYear={[Function]}
        />
      </div>
      <div
        className="spacer-bottom-lg"
      />
    </div>
    <div
      className="comments"
    >
      <PageHeader
        bsClass="page-header"
      >
        Comments 
        <small>
          (
          36
           
          Comments
          )
        </small>
      </PageHeader>
      <CommentsList
        comments={
          Array [
            Object {
              "body": "Comment 1",
              "created_at": "<time datetime=\\"2018-08-10T05:04:07Z\\" data-local=\\"time-ago\\">August 10, 2018  5:04am</time>",
              "gravatar_url": "https://gravatar.com/avatar/88884e5cb90c02031f031bc4760fcf69?s=80&r=pg&d=identicon",
              "id": 14,
              "user_name": "bluz71",
              "user_slug": "bluz71",
            },
            Object {
              "body": "Comment 2",
              "created_at": "<time datetime=\\"2018-08-04T01:14:51Z\\" data-local=\\"time-ago\\">August  4, 2018  1:14am</time>",
              "gravatar_url": "https://gravatar.com/avatar/5680ecac049040ce9f0a29c88b00df85?s=80&r=pg&d=identicon",
              "id": 10,
              "user_name": "jaron_barton50",
              "user_slug": "jaron_barton50",
            },
            Object {
              "body": "Comment 3",
              "created_at": "<time datetime=\\"2018-07-19T14:29:55Z\\" data-local=\\"time-ago\\">July 19, 2018  2:29pm</time>",
              "gravatar_url": "https://gravatar.com/avatar/69625f37b2fd5343d3d3439911d4e10f?s=80&r=pg&d=identicon",
              "id": 12,
              "user_name": "elise2",
              "user_slug": "elise2",
            },
            Object {
              "body": "Comment 4",
              "created_at": "<time datetime=\\"2018-07-16T19:29:24Z\\" data-local=\\"time-ago\\">July 16, 2018  7:29pm</time>",
              "gravatar_url": "https://gravatar.com/avatar/a38ce926222e8aee3f02999d435df4c4?s=80&r=pg&d=identicon",
              "id": 8,
              "user_name": "charley12",
              "user_slug": "charley12",
            },
            Object {
              "body": "Comment 5",
              "created_at": "<time datetime=\\"2018-06-15T01:54:55Z\\" data-local=\\"time-ago\\">June 15, 2018  1:54am</time>",
              "gravatar_url": "https://gravatar.com/avatar/7f14bdf8047c0883f230002a4bd2eff3?s=80&r=pg&d=identicon",
              "id": 11,
              "user_name": "austyn47",
              "user_slug": "austyn47",
            },
            Object {
              "body": "Comment 6",
              "created_at": "<time datetime=\\"2018-06-13T08:13:57Z\\" data-local=\\"time-ago\\">June 13, 2018  8:13am</time>",
              "gravatar_url": "https://gravatar.com/avatar/c4181331d29753fdc2d34d1995de937e?s=80&r=pg&d=identicon",
              "id": 6,
              "user_name": "ila_lebsack50",
              "user_slug": "ila_lebsack50",
            },
            Object {
              "body": "Comment 7",
              "created_at": "<time datetime=\\"2018-05-28T22:58:43Z\\" data-local=\\"time-ago\\">May 28, 2018 10:58pm</time>",
              "gravatar_url": "https://gravatar.com/avatar/693c529d5a6dc14404d15bd94944ab9d?s=80&r=pg&d=identicon",
              "id": 4,
              "user_name": "santos29",
              "user_slug": "santos29",
            },
            Object {
              "body": "Comment 8",
              "created_at": "<time datetime=\\"2018-04-19T10:55:23Z\\" data-local=\\"time-ago\\">April 19, 2018 10:55am</time>",
              "gravatar_url": "https://gravatar.com/avatar/5680ecac049040ce9f0a29c88b00df85?s=80&r=pg&d=identicon",
              "id": 1,
              "user_name": "jaron_barton50",
              "user_slug": "jaron_barton50",
            },
            Object {
              "body": "Comment 9",
              "created_at": "<time datetime=\\"2018-03-08T21:26:37Z\\" data-local=\\"time-ago\\">March  8, 2018  9:26pm</time>",
              "gravatar_url": "https://gravatar.com/avatar/6020481dd560c0383f9ef2b2ae8a0e0d?s=80&r=pg&d=identicon",
              "id": 5,
              "user_name": "yasmeen91",
              "user_slug": "yasmeen91",
            },
            Object {
              "body": "Comment 10",
              "created_at": "<time datetime=\\"2018-02-06T15:58:36Z\\" data-local=\\"time-ago\\">February  6, 2018  3:58pm</time>",
              "gravatar_url": "https://gravatar.com/avatar/ecd56b04440922d9d4227b80b360f1a6?s=80&r=pg&d=identicon",
              "id": 13,
              "user_name": "abbie35",
              "user_slug": "abbie35",
            },
            Object {
              "body": "Comment 11",
              "created_at": "<time datetime=\\"2018-01-18T21:28:48Z\\" data-local=\\"time-ago\\">January 18, 2018  9:28pm</time>",
              "gravatar_url": "https://gravatar.com/avatar/6020481dd560c0383f9ef2b2ae8a0e0d?s=80&r=pg&d=identicon",
              "id": 2,
              "user_name": "yasmeen91",
              "user_slug": "yasmeen91",
            },
            Object {
              "body": "Comment 12",
              "created_at": "<time datetime=\\"2018-01-06T18:46:18Z\\" data-local=\\"time-ago\\">January  6, 2018  6:46pm</time>",
              "gravatar_url": "https://gravatar.com/avatar/9c882241f6804414432bcb7c6fbb0dce?s=80&r=pg&d=identicon",
              "id": 3,
              "user_name": "michele_douglas54",
              "user_slug": "michele_douglas54",
            },
            Object {
              "body": "Comment 13",
              "created_at": "<time datetime=\\"2017-12-24T10:59:43Z\\" data-local=\\"time-ago\\">December 24, 2017 10:59am</time>",
              "gravatar_url": "https://gravatar.com/avatar/2d42930970262a4d6da8a839ef4cbed5?s=80&r=pg&d=identicon",
              "id": 9,
              "user_name": "roberto_zboncak62",
              "user_slug": "roberto_zboncak62",
            },
            Object {
              "body": "Comment 14",
              "created_at": "<time datetime=\\"2017-12-22T16:18:14Z\\" data-local=\\"time-ago\\">December 22, 2017  4:18pm</time>",
              "gravatar_url": "https://gravatar.com/avatar/4ecef4e31dc7b2bf3ac0117f3df7d1fb?s=80&r=pg&d=identicon",
              "id": 7,
              "user_name": "emily_spencer92",
              "user_slug": "emily_spencer92",
            },
            Object {
              "body": "Comment 15",
              "created_at": "<time datetime=\\"2018-08-10T05:04:07Z\\" data-local=\\"time-ago\\">August 10, 2018  5:04am</time>",
              "gravatar_url": "https://gravatar.com/avatar/88884e5cb90c02031f031bc4760fcf69?s=80&r=pg&d=identicon",
              "id": 14,
              "user_name": "bluz71",
              "user_slug": "bluz71",
            },
            Object {
              "body": "Comment 16",
              "created_at": "<time datetime=\\"2018-08-04T01:14:51Z\\" data-local=\\"time-ago\\">August  4, 2018  1:14am</time>",
              "gravatar_url": "https://gravatar.com/avatar/5680ecac049040ce9f0a29c88b00df85?s=80&r=pg&d=identicon",
              "id": 10,
              "user_name": "jaron_barton50",
              "user_slug": "jaron_barton50",
            },
            Object {
              "body": "Comment 17",
              "created_at": "<time datetime=\\"2018-07-19T14:29:55Z\\" data-local=\\"time-ago\\">July 19, 2018  2:29pm</time>",
              "gravatar_url": "https://gravatar.com/avatar/69625f37b2fd5343d3d3439911d4e10f?s=80&r=pg&d=identicon",
              "id": 12,
              "user_name": "elise2",
              "user_slug": "elise2",
            },
            Object {
              "body": "Comment 18",
              "created_at": "<time datetime=\\"2018-07-16T19:29:24Z\\" data-local=\\"time-ago\\">July 16, 2018  7:29pm</time>",
              "gravatar_url": "https://gravatar.com/avatar/a38ce926222e8aee3f02999d435df4c4?s=80&r=pg&d=identicon",
              "id": 8,
              "user_name": "charley12",
              "user_slug": "charley12",
            },
            Object {
              "body": "Comment 19",
              "created_at": "<time datetime=\\"2018-06-15T01:54:55Z\\" data-local=\\"time-ago\\">June 15, 2018  1:54am</time>",
              "gravatar_url": "https://gravatar.com/avatar/7f14bdf8047c0883f230002a4bd2eff3?s=80&r=pg&d=identicon",
              "id": 11,
              "user_name": "austyn47",
              "user_slug": "austyn47",
            },
            Object {
              "body": "Comment 20",
              "created_at": "<time datetime=\\"2018-06-13T08:13:57Z\\" data-local=\\"time-ago\\">June 13, 2018  8:13am</time>",
              "gravatar_url": "https://gravatar.com/avatar/c4181331d29753fdc2d34d1995de937e?s=80&r=pg&d=identicon",
              "id": 6,
              "user_name": "ila_lebsack50",
              "user_slug": "ila_lebsack50",
            },
            Object {
              "body": "Comment 21",
              "created_at": "<time datetime=\\"2018-05-28T22:58:43Z\\" data-local=\\"time-ago\\">May 28, 2018 10:58pm</time>",
              "gravatar_url": "https://gravatar.com/avatar/693c529d5a6dc14404d15bd94944ab9d?s=80&r=pg&d=identicon",
              "id": 4,
              "user_name": "santos29",
              "user_slug": "santos29",
            },
            Object {
              "body": "Comment 22",
              "created_at": "<time datetime=\\"2018-04-19T10:55:23Z\\" data-local=\\"time-ago\\">April 19, 2018 10:55am</time>",
              "gravatar_url": "https://gravatar.com/avatar/5680ecac049040ce9f0a29c88b00df85?s=80&r=pg&d=identicon",
              "id": 1,
              "user_name": "jaron_barton50",
              "user_slug": "jaron_barton50",
            },
            Object {
              "body": "Comment 23",
              "created_at": "<time datetime=\\"2018-03-08T21:26:37Z\\" data-local=\\"time-ago\\">March  8, 2018  9:26pm</time>",
              "gravatar_url": "https://gravatar.com/avatar/6020481dd560c0383f9ef2b2ae8a0e0d?s=80&r=pg&d=identicon",
              "id": 5,
              "user_name": "yasmeen91",
              "user_slug": "yasmeen91",
            },
            Object {
              "body": "Comment 24",
              "created_at": "<time datetime=\\"2017-11-04T04:31:56Z\\" data-local=\\"time-ago\\">November  4, 2017  4:31am</time>",
              "gravatar_url": "https://gravatar.com/avatar/7623bc3892c4bf9c2571447e23ec565f?s=80&r=pg&d=identicon",
              "id": 29998,
              "user_name": "leonel_kertzmann31",
              "user_slug": "leonel_kertzmann31",
            },
            Object {
              "body": "Last Comment 25",
              "created_at": "<time datetime=\\"2017-11-01T12:26:44Z\\" data-local=\\"time-ago\\">November  1, 2017 12:26pm</time>",
              "gravatar_url": "https://gravatar.com/avatar/188deba48d739c0287b061713d2c16be?s=80&r=pg&d=identicon",
              "id": 30003,
              "user_name": "simeon59",
              "user_slug": "simeon59",
            },
          ]
        }
        shortHeader={true}
      />
    </div>
  </Col>
</Row>
`;
